core/src/services/routing.js

```diff
@@ -93,11 +93,27 @@ export const isNotSameDomain = (config, component) => {
       componentData.previousNodeValues.viewUrl
     );
     const nextUrl = getUrlWithoutHash(componentData.viewUrl);
-    return previousUrl != nextUrl;
+    if (previousUrl === nextUrl) {
+      return false;
+    } else if (
+      componentData.viewUrl.startsWith('http://') ||
+      componentData.viewUrl.startsWith('https://')
+    ) {
+      const iFrameDomain = getDomainFromUrl(config.iframe.src);
+      const nextUrlDomain = getDomainFromUrl(nextUrl);
+      return iFrameDomain != nextUrlDomain;
+    } else {
+      return true;
+    }
   }
   return true;
 };
 
+const getDomainFromUrl = url => {
```

Comment:
We should be able to detect multiple SPAs under one domain, i.e:
`https://example.com/foo/` <-- root of one SPA
`https://example.com/bar/` <-- root of another SPA

Reply:
I guess we do not have a concept / configuration option for that, since you would be currently only able to detect hash based SPAs, path based are undetectable by only the url. In my opinion this is a followup feature, which requires grooming.